cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/ArtifactRepository.java

```diff
@@ -735,10 +735,10 @@ public void deleteArtifact(Id.Artifact artifactId) throws Exception {
     } else {
       authorizer.enforce(artifactId.toEntityId(), principal, Action.ADMIN);
     }
-    // revoke all privileges on the artifact
-    authorizer.revoke(artifactId.toEntityId());
     artifactStore.delete(artifactId);
     metadataStore.removeMetadata(artifactId);
+    // revoke all privileges on the artifact
+    authorizer.revoke(artifactId.toEntityId());
```

Comment:
can you add a comment here as to why we do it at the end? See DatasetInstanceService for the contents of that comment.